site/_wiki/Documentation/ConfigurationGuide.md

```diff
@@ -0,0 +1,418 @@
+
+There is more to detection than this (such as attributes), but 99.9% of use
+cases can be covered with the above.
+
+### ID's and report lengths
+
+At the very minimum, your configuration should have a non-null `VendorID` and `ProductID`.
+
+If your tablet has multiple interfaces, it is very likely only 1 of them
+contains the tablet data stream. This is where `InputReportLength` and
+`OutputReportLength` comes in.
+
+Even if your tablet only has a single interface, it is often a good idea to
+specify at least one of the lengths as non-zero, and only define them as `null`
+if you know what you're doing
+
+> If you specify report lengths as null, it can be hard to fix any future
+  conflicts that may arise without knowing what values your specific tablet
+  has. If you do decide to specify them as null, please make sure to include a
+  OpenTabletDriver diagnostics with the pull request, or include the lengths in
+  the commit description.
+
+### Picking the Parser {#parser}
+
+Picking the parser is the most important part of ensuring your tablet functions
+correctly. Picking the wrong parser can result in missing or (sometimes subtly)
+broken features.
+
+You must use a class which implements the `Parse(byte[])` function.
+
+How to write a parser from scratch is out of scope of this guide, but if you
+find that a parser you're using needs a code change to work, it is usually a
+better idea to make a new parser that builds off it instead.
+
+To see which functionality is supported by a parser, please see the code for
+the parsers return paths in its `Parse(byte[])` function. An example is
+provided below.
+
+#### `XP_PenReportParser` example
+
+At the time of this writing, [`XP_PenReportParser`] can return 5 different
+types of reports (not in order)
+
+[`XP_PenReportParser`]: https://github.com/OpenTabletDriver/OpenTabletDriver/blob/f14a184fef67a52637ab494497a151044bafd553/OpenTabletDriver.Configurations/Parsers/XP_Pen/XP_PenReportParser.cs#L7-L21
+
+##### `OutOfRangeReport`
+
+Fairly selfdescript, it doesn't actually report any specific values to the
+OS but rather 'resets' the pen out.
+
+##### `XP_PenAuxReport`:
+
+Reports `IAuxReport`, an aux-buttons-only report, ie. no position coordinates.
+
+##### `XP_PenTabletReport`
+
+Reports the following reports:
+
+- `ITabletReport`
+  - The pen's pressure
+  - Pen buttons
+  - It is an `IAbsolutePositionReport` that also reports the following:
+    - The 2-point vector of the pen
+- `ITiltReport`
+  - The 2-point vector of the pen's tilt
+- `IEraserReport`
+  - Whether the eraser is active or not
+
+##### `XP_PenTabletOverflowReport`
+
+An almost identical version of the above `XP_PenTabletReport` except with
+larger ranges for the position coordinate and the data structure adjusted
+accordingly.
+
+##### `TabletReport`
+
+A barebones tablet report that only reports an `ITabletReport`
+
+### Init strings (Feature/OutputInitReport, InitializationStrings)
```

Comment:
I don't think we really have added anything to this doc to indicate having the right initialization process?

Maybe add something somewhere (before deciding for a parser) that you should be able to see raw data in the debugger change when you use the tablet and that they'll be no failed to init messages in the debugger. (provide examples from feature/output failure? init strings do not have a message, so an example can't be provided)